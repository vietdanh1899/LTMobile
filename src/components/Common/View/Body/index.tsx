import React, { PureComponent } from 'react';
import { StyleSheet, SafeAreaView } from 'react-native';
import { withTheme, ThemeProps } from 'react-native-elements';
import QuickView, { QuickViewProps } from '../QuickView';

export interface BodyProps extends QuickViewProps {
  theme?: any;
  secondary?: boolean;
  primary?: boolean;
  fullWidth?: boolean;
  fullHeight?: boolean;
  fullView?: boolean;
}

class Body extends PureComponent<BodyProps> {
  static defaultProps = {
    primary: true,
  };

  render() {
    const {
      backgroundColor: backgroundColorProp,
      paddingHorizontal: paddingHorizontalProp,
      secondary,
      fullView,
      fullWidth,
      fullHeight,
      theme,
      style: styleProp,
      children,
      ...otherProps
    } = this.props;

    let backgroundColor = backgroundColorProp
      || (secondary ? theme.colors.bgColorSecondary : theme.colors.bgColor);
    const paddingHorizontal = fullWidth || fullView
      ? 0
      : paddingHorizontalProp || theme.bodyPaddingHorizontal;

    const { backgroundImage } = this.props;
    if (backgroundImage) backgroundColor = 'transparent';

    const style = StyleSheet.flatten([
      { backgroundColor, paddingHorizontal, flex: 1 },
      styleProp,
    ]);

    if (fullHeight || fullView) {
      return (
        <QuickView testID="FullHeightBodyView" {...otherProps} style={style}>
          {children}
        </QuickView>
      );
    }

    return (
      <SafeAreaView testID="SafeAreaBodyView" style={{ flex: 1 }}>
        <QuickView {...otherProps} style={style}>
          {children}
        </QuickView>
      </SafeAreaView>
    );
  }
}

export default withTheme(
  Body as React.ComponentType<BodyProps & ThemeProps<any>>,
);
