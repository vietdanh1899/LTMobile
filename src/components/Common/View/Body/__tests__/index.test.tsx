import React from 'react';
import {
  basicTest, EnumComponent, getComponent,
} from '@utils/testHelper';
import { lightTheme } from '@themes';
import Body from '..';

const target = <Body />;
const type = EnumComponent.HOC;

describe('Body Component', () => {
  basicTest(target, type, 2);

  it('should apply all custom props', () => {
    /**
     * primary, secondary
     */
    const props1 = getComponent(target, { primary: true }, type).props();
    expect(props1.children.props.style.backgroundColor).toBe(lightTheme.colors.bgColor);

    const props2 = getComponent(target, { secondary: true }, type).props();
    expect(props2.children.props.style.backgroundColor).toBe(lightTheme.colors.bgColorSecondary);

    /**
     * fullWidth, fullHeight, fullView
     */
    const props3 = getComponent(target, { fullWidth: true }, type).props();
    expect(props3.children.props.style.paddingHorizontal).toBe(0);
    expect(props3.testID).toBe('SafeAreaBodyView');

    const props4 = getComponent(target, { fullHeight: true }, type).props();
    expect(props4.testID).toBe('FullHeightBodyView');
    expect(props4.style.paddingHorizontal).toBe(lightTheme.bodyPaddingHorizontal);

    const props5 = getComponent(target, { fullView: true }, type).props();
    expect(props5.testID).toBe('FullHeightBodyView');
    expect(props5.style.paddingHorizontal).toBe(0);
  });
});
