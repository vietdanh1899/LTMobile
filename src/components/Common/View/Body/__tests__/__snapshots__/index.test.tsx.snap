// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Component should render without issues 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "flex": 1,
    }
  }
  testID="SafeAreaBodyView"
>
  <QuickView
    column={true}
    primary={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "paddingHorizontal": 18,
      }
    }
    touchableComponent="TouchableOpacity"
  />
</ForwardRef(SafeAreaView)>
`;
